File: framec_tests/csharp/SimpleHandlerCalls/SimpleHandlerCallsTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using NUnit.Framework;

namespace SimpleHandlerCalls
{
    class SimpleHandlerCallsController : SimpleHandlerCalls
    {
        public SimpleHandlerCallsController() : base()
        {
        }
    }
    [TestFixture]
    public class SimpleHandlerCallsTest
    {
        [Test]
        public void TestSimpleCall()
        {
            SimpleHandlerCallsController sm = new SimpleHandlerCallsController();
            sm.C();
            Assert.AreEqual("1", sm.state_info());
        }

        /* Test that a handler call terminates the current handler. */
        [Test]
        public void TestCallsTerminateHandler()
        {
            SimpleHandlerCallsController sm = new SimpleHandlerCallsController();
            sm.D();
            Assert.AreEqual("2", sm.state_info());

            sm = new SimpleHandlerCallsController();
            sm.E();
            Assert.AreEqual("2", sm.state_info());
        }

    }
}

File: framec_tests/cpp/simple_handler_calls/simple_handler_calls_test.cpp
#include <gtest/gtest.h>
#include "simple_handler_calls.cpp"

class SimpleHandlerCallsController : public SimpleHandlerCalls {
public:
    SimpleHandlerCallsController() : SimpleHandlerCalls() {}
};

TEST(SimpleHandlerCallsTest, TestSimpleCall) {
    SimpleHandlerCallsController sm;
    sm.C();
    ASSERT_EQ(sm.state_info(), "1");
}

TEST(SimpleHandlerCallsTest, TestCallsTerminateHandler) {
    SimpleHandlerCallsController sm;
    sm.D();
    ASSERT_EQ(sm.state_info(), "2");

    sm = SimpleHandlerCallsController();
    sm.E();
    ASSERT_EQ(sm.state_info(), "2");
}

int main(int argc, char** argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
